modules/builtin/src/backend/variables/boolean.ts

```diff
@@ -17,37 +17,45 @@ class BoxedBoolean implements BoxedVariable<boolean> {
   private _value?: boolean
   private _nbTurns?: number
 
-  constructor({ nbOfTurns, value }: BoxedVarContructor<boolean | string>) {
-    if (typeof value === 'boolean') {
-      this._value = value
-      this._confidence = 1
-    } else if (typeof value === 'string') {
-      this._value = Boolean(value)
-      this._confidence = 0.5
+  constructor({ nbOfTurns, value, confidence }: BoxedVarContructor<boolean>) {
+    if (value) {
+      this._nbTurns = nbOfTurns
+      this.trySet(value, confidence)
     }
-
-    this._nbTurns = nbOfTurns
   }
 
-  get value() {
-    return this._value
+  get value(): boolean {
+    return this._nbTurns !== 0 ? this._value : undefined
   }
 
-  set value(val) {
-    this._value = val
+  set value(newValue: boolean) {
+    this.trySet(newValue, 1)
   }
 
-  trySet(val: boolean, confidence: number) {
-    this._value = val
-    this._confidence = confidence
+  trySet(value: boolean, confidence: number) {
+    try {
+      if (typeof value === 'boolean') {
+        this._value = value
+        this._confidence = confidence
+      } else {
+        this._value = Boolean(value)
+        this._confidence = 0.5
```

Comment:
That conversion is not good, because Boolean('false') will return `true`. You could use the library `yn` to parse the value. If it returns true or false, then set the confidence to 1, otherwise 0